src/pages/page.js

```diff
@@ -2,13 +2,22 @@ import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 import Page from "../components/page";
 import { useFirestore } from "../context/firestoreContext";
+import { IoShareOutline } from "react-icons/io5" ; 
+import { ShareAdmin } from "../components/share";
+import { BsAlignBottom } from "react-icons/bs";
+import { useAdmin } from "../context/adminContext";
+
 
 export default function UserPage() {
   const { userId } = useParams();
   const { getUserDoc } = useFirestore();
   const [loading, setLoading] = useState(true);
 
   const [data, setData] = useState(null);
+  const [ buttonAnimate , setButtonAnimate ] = useState(false)
+  
+  const { state } = useAdmin();
+  const { username } = state;
```

Comment:
useAdmin hook will return the information of who is logged in , but in this case we do not need the username of the logged in person instead the username of the person we are currently visiting the page of. and that can be done by looking the userId parameter of the url . which is already done by userParams ,